src/lib.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+//!  This example shows you how to define ops in Rust and then call them from
+//!  JavaScript.
+// https://github.com/denoland/deno/blob/main/core/examples/hello_world.rs
+// https://github.com/denoland/deno/blob/main/core/examples/http_bench_json_ops.rs
+
+use deno_core::v8;
+use deno_core::JsRuntime;
+use deno_core::RuntimeOptions;
+use pgx::*;
+
+pg_module_magic!();
+
+// https://github.com/denoland/deno/blob/main/core/examples/eval_js_value.rs
+fn eval(context: &mut JsRuntime, code: &str) -> Result<serde_json::Value, String> {
+    let res = context.execute_script("<anon>", code);
+    match res {
+        Ok(global) => {
+            let scope = &mut context.handle_scope();
+            let local = v8::Local::new(scope, global);
+            // Deserialize a `v8` object into a Rust type using `serde_v8`,
+            // in this case deserialize to a JSON `Value`.
+            let deserialized_value = serde_v8::from_v8::<serde_json::Value>(scope, local);
+
+            match deserialized_value {
+                Ok(value) => Ok(value),
+                Err(err) => Err(format!("Cannot deserialize value: {:?}", err)),
+            }
+        }
+        Err(err) => Err(format!("Evaling error: {:?}", err)),
+    }
+}
```

Comment:
This is the key function, which evaluates a JS function. this is called from the extension below. eg:

```
create extension pldeno;
select pldeno('var x = 1; x + 1;');
```

Deno wraps the v8 engine, which means you can't return on the top level (`select pldeno('return "hello";')`). Instead I think we need to add some [ops](https://github.com/denoland/deno/blob/609c359dd448f2a7736ba1d93e94e1feff6acadc/core/examples/hello_world.rs#L26) to the runtime to call rust functions (?) which will operate in the PG space

Reply:
I extended this to support functions here: https://github.com/supabase/pldeno-test/tree/bs/pldeno-lang-ext

Still pretty limited, e.g. can't pass args.